ProjectHive/ProjectHive.Services.NotificationsAPI/Controllers/NotificationController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using ProjectHive.Services.NotificationsAPI.Dto;
+using ProjectHive.Services.NotificationsAPI.Models;
+using ProjectHive.Services.NotificationsAPI.Services;
+using System.Net.WebSockets;
+
+namespace ProjectHive.Services.NotificationsAPI.Controllers
+{
+    public class NotificationController : Controller
+    {
+        private readonly IWebSocketHandler _webSocketHandler;
+        private readonly IMapper _mapper;
+        public NotificationController(IMapper mapper)
+        {
+            _webSocketHandler = new WebSocketHandler();
+            _mapper = mapper;
```

Comment:
Use dependency injection to inject IWebSocketHandler into the controller. This will improve testability and follow the ASP.NET Core design guidelines.

Reply:
Make sure to register the WebSocketHandler as a service